src/coreclr/vm/multicorejitplayer.cpp

```diff
@@ -1349,53 +1106,149 @@ HRESULT MulticoreJitProfilePlayer::PlayProfile()
 
     while ((SUCCEEDED(hr)) && (nSize > sizeof(unsigned)))
     {
-        unsigned data   = * (const unsigned *) pBuffer;
-        unsigned rcdLen = data & 0xFFFFFF;
-        unsigned rcdTyp = data >> 24;
+        unsigned data1 = * (const unsigned *) pBuffer;
+        unsigned data2 = 0;
+        unsigned rcdTyp = data1 >> RECORD_TYPE_OFFSET;
+        unsigned rcdLen = 0;
+        
+        if (rcdTyp == MULTICOREJIT_MODULE_RECORD_ID)
+        {
+            rcdLen = data1 & 0xFFFFFF;    
+        }
+        else if (rcdTyp == MULTICOREJIT_MODULEINF_RECORD_ID)
+        {
+            rcdLen = sizeof(unsigned);
+        }
+        else if (rcdTyp == MULTICOREJIT_METHODINF_RECORD_ID)
+        {
+            data2 = * (((const unsigned *) pBuffer) + 1);
+            rcdLen = data2 >> SIGNATURE_LENGTH_OFFSET;
+        }
+        else
+        {
+            hr = COR_E_BADIMAGEFORMAT;
+            break;
+        }
 
         if ((rcdLen > nSize) || (rcdLen & 3)) // Better DWORD align
         {
             hr = COR_E_BADIMAGEFORMAT;
+            break;
         }
-        else
+
+        if (rcdTyp == MULTICOREJIT_MODULE_RECORD_ID)
+        {
+            const ModuleRecord * pRec = (const ModuleRecord * ) pBuffer;
+
+            if (((unsigned)(pRec->lenModuleName
+                + pRec->lenAssemblyName
+                ) > (rcdLen - sizeof(ModuleRecord))) ||
+                (m_moduleCount >= m_headerModuleCount))
+            {
+                hr = COR_E_BADIMAGEFORMAT;
+            }
+            else
+            {
+                hr = HandleModuleRecord(pRec);
+            }
+        }
+        else if (rcdTyp == MULTICOREJIT_MODULEINF_RECORD_ID)
         {
-            if (rcdTyp == MULTICOREJIT_MODULE_RECORD_ID)
+            unsigned moduleIndex = data1 & (MAX_MODULES - 1);
+            unsigned level = (data1 & ((MAX_MODULE_LEVELS - 1) << MODULE_LEVEL_OFFSET)) >> MODULE_LEVEL_OFFSET;
+
+            hr = HandleModuleInfoRecord(moduleIndex, level);
+        }
+        else if (rcdTyp == MULTICOREJIT_METHODINF_RECORD_ID)
+        {
+            // Find all subsequent methods and jit/load them reversed if reversed order is required
+            bool reversedOrder = true;
+
+            if (reversedOrder)
```

Comment:
Nit: if this is never false we may as well eliminate the if statement

Reply:
This actually would be part of the next PR, that's why initially I left it here even if it was dead code. Removed from this PR